src/server.ts

```
@@ -102,9 +134,65 @@ export class CodexMcpServer {
+      return timeoutMs;
+    }
+
+    return this.toolTimeoutMs;
+  }
+
+  private async withTimeout<T>(
+    operation: Promise<T>,
+    timeoutMs: number,
+    controller?: AbortController
+  ): Promise<T> {
+    let timeoutHandle: ReturnType<typeof setTimeout> | undefined;
```

ESLint

[error] 'setTimeout' is not defined.

(no-undef)